tonic-types/src/lib.rs

```diff
@@ -13,8 +23,391 @@
 #![doc(html_root_url = "https://docs.rs/tonic-types/0.6.0")]
 #![doc(issue_tracker_base_url = "https://github.com/hyperium/tonic/issues/")]
 
-mod pb {
+use prost::{DecodeError, Message};
+use prost_types::Any;
+use tonic::{codegen::Bytes, Code};
+
+/// Useful protobuf types
+pub mod pb {
     include!(concat!(env!("OUT_DIR"), "/google.rpc.rs"));
 }
 
 pub use pb::Status;
+
+mod error_details;
+mod error_details_vec;
+mod std_messages;
+
+pub use std_messages::*;
+
+pub use error_details::ErrorDetails;
+
+pub use error_details_vec::ErrorDetail;
+
+trait IntoAny {
+    fn into_any(self) -> Any;
+}
+
+trait FromAny {
+    fn from_any(any: Any) -> Result<Self, DecodeError>
+    where
+        Self: Sized;
+}
+
+/// Used to implement associated functions and methods on `tonic::Status`, that
+/// allow the addition and extraction of standard error details.
+pub trait WithErrorDetails {
+    /// Generates a `tonic::Status` with error details obtained from an
+    /// [`ErrorDetails`] struct.
+    /// # Examples
+    ///
+    /// ```
+    /// use tonic::{Code, Status};
+    /// use tonic_types::{ErrorDetails, WithErrorDetails};
+    ///
+    /// let status = Status::with_error_details(
+    ///     Code::InvalidArgument,
+    ///     "bad request",
+    ///     ErrorDetails::with_bad_request_violation("field", "description"),
+    /// );
+    /// ```
+    fn with_error_details(
+        code: tonic::Code,
+        message: impl Into<String>,
+        details: ErrorDetails,
+    ) -> tonic::Status;
+
+    /// Generates a `tonic::Status` with error details provided in a vector of
+    /// [`ErrorDetail`] enums.
+    /// # Examples
+    ///
+    /// ```
+    /// use tonic::{Code, Status};
+    /// use tonic_types::{BadRequest, WithErrorDetails};
+    ///
+    /// let status = Status::with_error_details_vec(
+    ///     Code::InvalidArgument,
+    ///     "bad request",
+    ///     vec![
+    ///         BadRequest::with_violation("field", "description").into(),
+    ///     ]
+    /// );
+    /// ```
+    fn with_error_details_vec(
+        code: tonic::Code,
```

Comment:
```suggestion
        code: Code,
```
I would use this style since this is already a tonic crate.

Reply:
Ok, I addressed this in the adjustments commit